### PR TITLE
Feat: add tranpile option to format

### PR DESCRIPTION
**Pull request by z3z1ma**

### CODE REVIEW

**Comment by tobymao** [comment]
as discussed, we need to edit the model dialect if it exists

File: sqlmesh/core/context.py [hunk 1]
```diff
@@ -628,8 +628,16 @@ def format(self) -> None:
                 expressions = parse(
                     file.read(), default_dialect=self.config_for_model(model).dialect
                 )
+                for prop in expressions[0].expressions:
+                    p = t.cast(exp.Property, prop)
```
Comment (tobymao) [inline sqlmesh/core/context.py:632]: i don't think you need this cast here do you?

File: sqlmesh/core/context.py [hunk 2]
```diff
@@ -628,8 +628,16 @@ def format(self) -> None:
                 expressions = parse(
                     file.read(), default_dialect=self.config_for_model(model).dialect
                 )
+                for prop in expressions[0].expressions:
```
Comment (tobymao) [inline sqlmesh/core/context.py:631]: this should only run if transpile right? otherwise don't touch it

File: sqlmesh/core/context.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/context.py]: yea let's remove it, it's not that useful in this context

Comment (z3z1ma) [reply sqlmesh/core/context.py]: @tobymao done

**Review by tobymao — APPROVED** [review]

**Pull request merged**